File: src/main/java/dev/misei/einfachml/repository/model/MSEData.java
package dev.misei.einfachml.repository.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.UUID;

@AllArgsConstructor
@Data
@Document
public class MSEData {
    UUID networkId;
    int epochHappened;
    Double error;
}

File: src/main/resources/static/classes/MSEData.js
export class MSEData {
    constructor(networkId, epochHappened, error) {
        this.networkId = networkId;  // UUID
        this.epochHappened = epochHappened;  // int
        this.error = error;  // double as number
    }

    static fromJson(jsonObj) {
        return new MSEData(
            jsonObj.networkId,
            jsonObj.epochHappened,
            jsonObj.error
        );
    }

    equals(other) {
        if (!(other instanceof MSEData)) {
            return false;
        }

        return (
            this.networkId === other.networkId &&
            this.epochHappened === other.epochHappened &&
            this.error === other.error
        );
    }
}